microcosm_pubsub/dispatcher.py

```diff
@@ -40,37 +40,29 @@ def handle_message(self, message, bound_handlers) -> MessageHandlingResult:
         Handle a message.
 
         """
-        # initialize a context for the current message
         with self.opaque.initialize(self.sqs_message_context, message):
-            # save elapsed time information to the context
+            handler = None
             with elapsed_time(self.opaque):
-                message_handling_result = MessageHandlingResult.invoke(
-                    func=self._handle_message,
-                    message=message,
-                    bound_handlers=bound_handlers,
-                )
-            # save elapsed time
-            message_handling_result.elapsed_time = self.opaque["elapsed_time"]
-            # log handler outcome
-            logger = self.choose_logger(message.handler)
-            message_handling_result.log(logger, self.opaque)
-        return message_handling_result
-
-    def _handle_message(self, message, bound_handlers) -> bool:
-        """
-        Handle a single message.
-
-        :raises: Nack, IgnoreMessage, SkipMessage, TTLExpired
-
-        """
-        # we might just want to ignore the message
-        self.validate_ttl()
-        content = self.validate_content(message)
-        handler = self.find_handler(message, bound_handlers)
-
-        # handle the message
-        result = handler(content)
-        return bool(result)
+                try:
+                    self.validate_ttl()
+                    self.validate_content(message)
+                    handler = self.find_handler(message, bound_handlers)
+                    instance = MessageHandlingResult.invoke(
+                        handler=self.wrap_handler(handler),
```

Comment:
We move the handler wrapping here so that the reference on L49 points to the actual handler (and not the wrapped one).